### PR TITLE
npctalk: add u_sell_item effect

### PR DESCRIPTION
**Pull request by mlangsdorf**
#### Summary
```SUMMARY: Infrastructure "npctalk: add u_sell_item effect"```

#### Purpose of change
Partially addresses #27397
Add an effect that removes an item from the player's inventory and adds it to the NPC's inventory.

#### Describe the solution
Basically the reverse of u_buy_item: find an item in the player's inventory, remove it, and add it to the NPC's inventory.

#### Describe alternatives you've considered
I couldn't figure out how to handle containers.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
jenkins rebuild

File: src/npctalk.cpp [hunk 1]
```diff
@@ -1940,6 +1940,32 @@ void talk_effect_fun_t::set_u_buy_item( const std::string &item_name, int cost,
...
+            p.i_add( old_item );
+        }
+        if( count == 1 ) {
+            //~ %1%s is the NPC name, %2$s is an item
+            popup( _( "You give %1$s a %2$s" ), p.name, old_item.tname() );
+        } else {
+            //~ %1%s is the NPC name, %2$d is a number of items, %3$s are items
+            popup( _( "You give %1$s %2$d %3$s" ), p.name, count, old_item.tname() );
+        }
+        u.cash -= cost;
```
Comment (GoLoT) [inline src/npctalk.cpp]: Shouldn't this be `u.cash += cost;` instead? Unless costs for items are to be set as negative values, of course.

Comment (mlangsdorf) [reply src/npctalk.cpp]: Fixed.

File: doc/NPCs.md [hunk 2]
```diff
@@ -533,6 +533,10 @@ Your character or the NPC will gain the trait.
 ### u_buy_item: item_string, (optional cost: cost_num, optional count: count_num, optional container: container_string)
 The NPC will give your character the item or count_num copies of the item, contained in container, and will remove `cost_num` from your character's cash if specified.  If cost isn't present, the NPC gives your character the item at no charge.
 
+### u_buy_item: item_string, (optional cost: cost_num, optional count: count_num)
+Your character will give the NPC the item or count_num copies of the item, and will remove `cost_num` from your character's cash if specified.  If cost isn't present, the your character gives the NPC the item at no charge.
```
Comment (ZhilkinSerg) [inline doc/NPCs.md]:
```suggestion
Your character will give the NPC the item or `count_num` copies of the item, and will remove `cost_num` from your character's cash if specified.  If cost isn't present, the your character gives the NPC the item at no charge.
```

File: doc/NPCs.md [hunk 3]
```diff
@@ -533,6 +533,10 @@ Your character or the NPC will gain the trait.
...
+This effect will fail if you do not have at least count_num copies of the item, so it should be checked with u_has_items.
```
Comment (ZhilkinSerg) [inline doc/NPCs.md]:
```suggestion
This effect will fail if you do not have at least `count_num` copies of the item, so it should be checked with `u_has_items`.
```

File: doc/NPCs.md (hunk 2 above)
Comment (mlangsdorf) [reply doc/NPCs.md]: Fixed, thanks for the review.

**Pull request merged**